axiom/nr_distributed_trace.c

```diff
@@ -488,7 +488,15 @@ void nr_distributed_trace_set_trace_id(nr_distributed_trace_t* dt,
 
   nr_free(dt->trace_id);
   if (trace_id) {
-    dt->trace_id = nr_strdup(trace_id);
+    int len = nr_strlen(trace_id);
+    if (len < NR_TRACE_ID_SIZE) {
+      int padding = NR_TRACE_ID_SIZE - len;
+      char* dest = (char*)(malloc)(NR_TRACE_ID_SIZE + 1);
```

Comment:
Why not using `nr_malloc` or `nr_zalloc`:
```suggestion
      char* dest = (char*)nr_malloc(NR_TRACE_ID_SIZE + 1);
```
`nr_(m|z)alloc` are guaranteed to succeed or crash. The code in this PR does not check  for `malloc`'s return value.